src/Codeception/Configuration.php

```diff
@@ -251,21 +253,37 @@ public static function config($configFile = null)
         }
 
         Autoload::addNamespace(self::$config['namespace'], self::supportDir());
-        self::loadBootstrap($config['settings']['bootstrap']);
+
+        if ($config['settings']['bootstrap']) {
+            $bootstrap = self::$config['settings']['bootstrap'];
+            Notification::deprecate("'settings: bootstrap: $bootstrap' option is deprecated! Replace it with: 'bootstrap: $bootstrap' (not under settings section). See https://bit.ly/2YrRzVc ");
+            try {
+                self::loadBootstrap($bootstrap, self::testsDir());
+            } catch (ConfigurationException $exception) {
+                Notification::deprecate("Bootstrap file ($bootstrap) is defined in configuration but can't be loaded. Disable 'settings: bootstrap:' configuration to remove this message");
+            }
+        }
+        self::loadBootstrap($config['bootstrap'], self::testsDir());
         self::loadSuites();
 
         return $config;
     }
 
-    protected static function loadBootstrap($bootstrap)
+    public static function loadBootstrap($bootstrap, $path)
     {
         if (!$bootstrap) {
             return;
         }
-        $bootstrap = self::$dir . DIRECTORY_SEPARATOR . self::$testsDir . DIRECTORY_SEPARATOR . $bootstrap;
-        if (file_exists($bootstrap)) {
-            include_once $bootstrap;
+
+        $bootstrap = codecept_is_path_absolute($bootstrap)
+            ? $bootstrap
+            : rtrim($path, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $bootstrap;
+
+
+        if (!file_exists($bootstrap)) {
+            throw new ConfigurationException("Bootstrap file $bootstrap can't be loaded");
         }
+        include_once $bootstrap;
```

Comment:
Why is this method using `include_once`? `Subscriber/Bootstrap.php` below used `require_once` before.
But I think that it would be better to use `require` rather than `require_once`, because boostrap files are more likely to contain linear code than class/file definitions and after this change it will be possible for multiple test suites to have the same absolute bootstrap file path.

Reply:
I think it's dangerous to switch this to `require` as this can produce fatal errors in current setups.
Bootstrap is usually used for autoloaders and I think that this should be enough to execute it once